### PR TITLE
Improve vApps properties section adding categories and sort order

### PR DESCRIPTION
**Pull request by nvazquez**
For deploy-as-is templates:
- Add category when available
- Sort properties in the order defined in the template

### CODE REVIEW

**Comment by rhtyd** [comment]
cc @shwstppr
LGTM
@blueorangutan package

**Review by rhtyd — APPROVED** [review]
LGTM, did not test against a vapp env but I'll test against primate-qa env for normal VM deployment. Can merge as soon as @shwstppr LGTMs

**Comment by rhtyd** [comment]
Regression test LGTM, tried to deploy a VM with some changes in templates and root disk size override

File: src/views/compute/DeployVM.vue [hunk 1]
```diff
@@ -1791,4 +1806,10 @@ export default {
   .form-item-hidden {
     display: none;
   }
+
+  .vapp-category {
+    background-color: lightblue;
```
Comment (shwstppr) [inline src/views/compute/DeployVM.vue:1811]: Screenshot from 2020-08-19 15-04-24
Can we use some standard primate color. Looks different to rest of the UI

**Review by shwstppr — CHANGES_REQUESTED** [review]
Seeing issue with one of the multi-disk ova while showing properties, configurations. WIll have a look myself. Will be better to club issues around vApps

> vue.runtime.esm.js?2b0e:619 [Vue warn]: Error in render: "TypeError: Cannot read property 'length' of undefined"
> 
> found in
> 
> ---> <Wizard> at src/views/compute/DeployVM.vue
>        <RouteView> at src/layouts/RouteView.vue... (1 recursive calls)
>          <Anonymous>
>            <ALayoutContent>
>              <Anonymous>
>                <ALayout>
>                  <Anonymous>
>                    <ALayout>
>                      <GlobalLayout> at src/components/page/GlobalLayout.vue
>                        <BasicLayout> at src/layouts/BasicLayout.vue
>                          <ALocaleProvider>
>                            <LocaleReceiver>
>                              <AConfigProvider>
>                                <App> at src/App.vue
>                                  <Root>
> warn @ vue.runtime.esm.js?2b0e:619
> logError @ vue.runtime.esm.js?2b0e:1884
> globalHandleError @ vue.runtime.esm.js?2b0e:1879
> handleError @ vue.runtime.esm.js?2b0e:1839
> Vue._render @ vue.runtime.esm.js?2b0e:3550
> updateComponent @ vue.runtime.esm.js?2b0e:4066
> get @ vue.runtime.esm.js?2b0e:4479
> run @ vue.runtime.esm.js?2b0e:4554
> flushSchedulerQueue @ vue.runtime.esm.js?2b0e:4310
> eval @ vue.runtime.esm.js?2b0e:1980
> flushCallbacks @ vue.runtime.esm.js?2b0e:1906
> Promise.then (async)
> timerFunc @ vue.runtime.esm.js?2b0e:1933
> nextTick @ vue.runtime.esm.js?2b0e:1990
> queueWatcher @ vue.runtime.esm.js?2b0e:4402
> update @ vue.runtime.esm.js?2b0e:4544
> Vue.$forceUpdate @ vue.runtime.esm.js?2b0e:3969
> handleChange @ Checkbox.js?d994:76
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> original._wrapper @ vue.runtime.esm.js?2b0e:6917
> vue.runtime.esm.js?2b0e:1888 TypeError: Cannot read property 'length' of undefined
>     at eval (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1446:76)
>     at Proxy.renderList (vue.runtime.esm.js?2b0e:2630)
>     at eval (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1281:49)
>     at Proxy.renderList (vue.runtime.esm.js?2b0e:2651)
>     at Proxy.render (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1258:41)
>     at VueComponent.Vue._render (vue.runtime.esm.js?2b0e:3548)
>     at VueComponent.updateComponent (vue.runtime.esm.js?2b0e:4066)
>     at Watcher.get (vue.runtime.esm.js?2b0e:4479)
>     at Watcher.run (vue.runtime.esm.js?2b0e:4554)
>     at flushSchedulerQueue (vue.runtime.esm.js?2b0e:4310)
> logError @ vue.runtime.esm.js?2b0e:1888
> globalHandleError @ vue.runtime.esm.js?2b0e:1879
> handleError @ vue.runtime.esm.js?2b0e:1839
> Vue._render @ vue.runtime.esm.js?2b0e:3550
> updateComponent @ vue.runtime.esm.js?2b0e:4066
> get @ vue.runtime.esm.js?2b0e:4479
> run @ vue.runtime.esm.js?2b0e:4554
> flushSchedulerQueue @ vue.runtime.esm.js?2b0e:4310
> eval @ vue.runtime.esm.js?2b0e:1980
> flushCallbacks @ vue.runtime.esm.js?2b0e:1906
> Promise.then (async)
> timerFunc @ vue.runtime.esm.js?2b0e:1933
> nextTick @ vue.runtime.esm.js?2b0e:1990
> queueWatcher @ vue.runtime.esm.js?2b0e:4402
> update @ vue.runtime.esm.js?2b0e:4544
> Vue.$forceUpdate @ vue.runtime.esm.js?2b0e:3969
> handleChange @ Checkbox.js?d994:76
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> original._wrapper @ vue.runtime.esm.js?2b0e:6917
> vue.runtime.esm.js?2b0e:619 [Vue warn]: Error in render: "TypeError: Cannot read property 'length' of undefined"
> 
> found in
> 
> ---> <Wizard> at src/views/compute/DeployVM.vue
>        <RouteView> at src/layouts/RouteView.vue... (1 recursive calls)
>          <Anonymous>
>            <ALayoutContent>
>              <Anonymous>
>                <ALayout>
>                  <Anonymous>
>                    <ALayout>
>                      <GlobalLayout> at src/components/page/GlobalLayout.vue
>                        <BasicLayout> at src/layouts/BasicLayout.vue
>                          <ALocaleProvider>
>                            <LocaleReceiver>
>                              <AConfigProvider>
>                                <App> at src/App.vue
>                                  <Root>
> warn @ vue.runtime.esm.js?2b0e:619
> logError @ vue.runtime.esm.js?2b0e:1884
> globalHandleError @ vue.runtime.esm.js?2b0e:1879
> handleError @ vue.runtime.esm.js?2b0e:1839
> Vue._render @ vue.runtime.esm.js?2b0e:3550
> updateComponent @ vue.runtime.esm.js?2b0e:4066
> get @ vue.runtime.esm.js?2b0e:4479
> run @ vue.runtime.esm.js?2b0e:4554
> flushSchedulerQueue @ vue.runtime.esm.js?2b0e:4310
> eval @ vue.runtime.esm.js?2b0e:1980
> flushCallbacks @ vue.runtime.esm.js?2b0e:1906
> Promise.then (async)
> timerFunc @ vue.runtime.esm.js?2b0e:1933
> nextTick @ vue.runtime.esm.js?2b0e:1990
> queueWatcher @ vue.runtime.esm.js?2b0e:4402
> update @ vue.runtime.esm.js?2b0e:4544
> Vue.$forceUpdate @ vue.runtime.esm.js?2b0e:3969
> handleChange @ Checkbox.js?d994:76
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> original._wrapper @ vue.runtime.esm.js?2b0e:6917
> vue.runtime.esm.js?2b0e:1888 TypeError: Cannot read property 'length' of undefined
>     at eval (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1446:76)
>     at Proxy.renderList (vue.runtime.esm.js?2b0e:2630)
>     at eval (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1281:49)
>     at Proxy.renderList (vue.runtime.esm.js?2b0e:2651)
>     at Proxy.render (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1258:41)
>     at VueComponent.Vue._render (vue.runtime.esm.js?2b0e:3548)
>     at VueComponent.updateComponent (vue.runtime.esm.js?2b0e:4066)
>     at Watcher.get (vue.runtime.esm.js?2b0e:4479)
>     at Watcher.run (vue.runtime.esm.js?2b0e:4554)
>     at flushSchedulerQueue (vue.runtime.esm.js?2b0e:4310)
> logError @ vue.runtime.esm.js?2b0e:1888
> globalHandleError @ vue.runtime.esm.js?2b0e:1879
> handleError @ vue.runtime.esm.js?2b0e:1839
> Vue._render @ vue.runtime.esm.js?2b0e:3550
> updateComponent @ vue.runtime.esm.js?2b0e:4066
> get @ vue.runtime.esm.js?2b0e:4479
> run @ vue.runtime.esm.js?2b0e:4554
> flushSchedulerQueue @ vue.runtime.esm.js?2b0e:4310
> eval @ vue.runtime.esm.js?2b0e:1980
> flushCallbacks @ vue.runtime.esm.js?2b0e:1906
> Promise.then (async)
> timerFunc @ vue.runtime.esm.js?2b0e:1933
> nextTick @ vue.runtime.esm.js?2b0e:1990
> queueWatcher @ vue.runtime.esm.js?2b0e:4402
> update @ vue.runtime.esm.js?2b0e:4544
> Vue.$forceUpdate @ vue.runtime.esm.js?2b0e:3969
> handleChange @ Checkbox.js?d994:76
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> original._wrapper @ vue.runtime.esm.js?2b0e:6917
> DeployVM.vue?7176:1718 
> vue.runtime.esm.js?2b0e:619 [Vue warn]: Error in render: "TypeError: Cannot read property 'length' of undefined"
> 
> found in
> 
> ---> <Wizard> at src/views/compute/DeployVM.vue
>        <RouteView> at src/layouts/RouteView.vue... (1 recursive calls)
>          <Anonymous>
>            <ALayoutContent>
>              <Anonymous>
>                <ALayout>
>                  <Anonymous>
>                    <ALayout>
>                      <GlobalLayout> at src/components/page/GlobalLayout.vue
>                        <BasicLayout> at src/layouts/BasicLayout.vue
>                          <ALocaleProvider>
>                            <LocaleReceiver>
>                              <AConfigProvider>
>                                <App> at src/App.vue
>                                  <Root>
> warn @ vue.runtime.esm.js?2b0e:619
> logError @ vue.runtime.esm.js?2b0e:1884
> globalHandleError @ vue.runtime.esm.js?2b0e:1879
> handleError @ vue.runtime.esm.js?2b0e:1839
> Vue._render @ vue.runtime.esm.js?2b0e:3550
> updateComponent @ vue.runtime.esm.js?2b0e:4066
> get @ vue.runtime.esm.js?2b0e:4479
> run @ vue.runtime.esm.js?2b0e:4554
> flushSchedulerQueue @ vue.runtime.esm.js?2b0e:4310
> eval @ vue.runtime.esm.js?2b0e:1980
> flushCallbacks @ vue.runtime.esm.js?2b0e:1906
> Promise.then (async)
> timerFunc @ vue.runtime.esm.js?2b0e:1933
> nextTick @ vue.runtime.esm.js?2b0e:1990
> queueWatcher @ vue.runtime.esm.js?2b0e:4402
> update @ vue.runtime.esm.js?2b0e:4544
> notify @ vue.runtime.esm.js?2b0e:730
> reactiveSetter @ vue.runtime.esm.js?2b0e:1055
> proxySetter @ vue.runtime.esm.js?2b0e:4631
> eval @ DeployVM.vue?7176:1719
> setTimeout (async)
> updateTemplateParameters @ DeployVM.vue?7176:1716
> updateFieldValue @ DeployVM.vue?7176:1142
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> Vue.$emit @ vue.runtime.esm.js?2b0e:3888
> updateTemplateIso @ TemplateIsoSelection.vue?9200:125
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> Vue.$emit @ vue.runtime.esm.js?2b0e:3888
> updateSelectionTemplateIso @ TemplateIsoRadioGroup.vue?903a:121
> change @ TemplateIsoRadioGroup.vue?7a8a:31
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> Vue.$emit @ vue.runtime.esm.js?2b0e:3888
> onRadioChange @ Group.js?89ee:92
> onChange @ Radio.js?d338:51
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> __emit @ BaseMixin.js?b488:37
> handleChange @ Checkbox.js?d994:78
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> original._wrapper @ vue.runtime.esm.js?2b0e:6917
> vue.runtime.esm.js?2b0e:1888 TypeError: Cannot read property 'length' of undefined
>     at eval (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1446:76)
>     at Proxy.renderList (vue.runtime.esm.js?2b0e:2630)
>     at eval (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1281:49)
>     at Proxy.renderList (vue.runtime.esm.js?2b0e:2651)
>     at Proxy.render (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1258:41)
>     at VueComponent.Vue._render (vue.runtime.esm.js?2b0e:3548)
>     at VueComponent.updateComponent (vue.runtime.esm.js?2b0e:4066)
>     at Watcher.get (vue.runtime.esm.js?2b0e:4479)
>     at Watcher.run (vue.runtime.esm.js?2b0e:4554)
>     at flushSchedulerQueue (vue.runtime.esm.js?2b0e:4310)
> logError @ vue.runtime.esm.js?2b0e:1888
> globalHandleError @ vue.runtime.esm.js?2b0e:1879
> handleError @ vue.runtime.esm.js?2b0e:1839
> Vue._render @ vue.runtime.esm.js?2b0e:3550
> updateComponent @ vue.runtime.esm.js?2b0e:4066
> get @ vue.runtime.esm.js?2b0e:4479
> run @ vue.runtime.esm.js?2b0e:4554
> flushSchedulerQueue @ vue.runtime.esm.js?2b0e:4310
> eval @ vue.runtime.esm.js?2b0e:1980
> flushCallbacks @ vue.runtime.esm.js?2b0e:1906
> Promise.then (async)
> timerFunc @ vue.runtime.esm.js?2b0e:1933
> nextTick @ vue.runtime.esm.js?2b0e:1990
> queueWatcher @ vue.runtime.esm.js?2b0e:4402
> update @ vue.runtime.esm.js?2b0e:4544
> notify @ vue.runtime.esm.js?2b0e:730
> reactiveSetter @ vue.runtime.esm.js?2b0e:1055
> proxySetter @ vue.runtime.esm.js?2b0e:4631
> eval @ DeployVM.vue?7176:1719
> setTimeout (async)
> updateTemplateParameters @ DeployVM.vue?7176:1716
> updateFieldValue @ DeployVM.vue?7176:1142
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> Vue.$emit @ vue.runtime.esm.js?2b0e:3888
> updateTemplateIso @ TemplateIsoSelection.vue?9200:125
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> Vue.$emit @ vue.runtime.esm.js?2b0e:3888
> updateSelectionTemplateIso @ TemplateIsoRadioGroup.vue?903a:121
> change @ TemplateIsoRadioGroup.vue?7a8a:31
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> Vue.$emit @ vue.runtime.esm.js?2b0e:3888
> onRadioChange @ Group.js?89ee:92
> onChange @ Radio.js?d338:51
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> __emit @ BaseMixin.js?b488:37
> handleChange @ Checkbox.js?d994:78
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> original._wrapper @ vue.runtime.esm.js?2b0e:6917
> DeployVM.vue?7176:1718 
> vue.runtime.esm.js?2b0e:619 [Vue warn]: Error in render: "TypeError: Cannot read property 'length' of undefined"
> 
> found in
> 
> ---> <Wizard> at src/views/compute/DeployVM.vue
>        <RouteView> at src/layouts/RouteView.vue... (1 recursive calls)
>          <Anonymous>
>            <ALayoutContent>
>              <Anonymous>
>                <ALayout>
>                  <Anonymous>
>                    <ALayout>
>                      <GlobalLayout> at src/components/page/GlobalLayout.vue
>                        <BasicLayout> at src/layouts/BasicLayout.vue
>                          <ALocaleProvider>
>                            <LocaleReceiver>
>                              <AConfigProvider>
>                                <App> at src/App.vue
>                                  <Root>
> warn @ vue.runtime.esm.js?2b0e:619
> logError @ vue.runtime.esm.js?2b0e:1884
> globalHandleError @ vue.runtime.esm.js?2b0e:1879
> handleError @ vue.runtime.esm.js?2b0e:1839
> Vue._render @ vue.runtime.esm.js?2b0e:3550
> updateComponent @ vue.runtime.esm.js?2b0e:4066
> get @ vue.runtime.esm.js?2b0e:4479
> run @ vue.runtime.esm.js?2b0e:4554
> flushSchedulerQueue @ vue.runtime.esm.js?2b0e:4310
> eval @ vue.runtime.esm.js?2b0e:1980
> flushCallbacks @ vue.runtime.esm.js?2b0e:1906
> Promise.then (async)
> timerFunc @ vue.runtime.esm.js?2b0e:1933
> nextTick @ vue.runtime.esm.js?2b0e:1990
> queueWatcher @ vue.runtime.esm.js?2b0e:4402
> update @ vue.runtime.esm.js?2b0e:4544
> Vue.$forceUpdate @ vue.runtime.esm.js?2b0e:3969
> setFields @ createBaseForm.js?bcc9:345
> setFieldsValue @ createBaseForm.js?bcc9:368
> updateFieldValue @ DeployVM.vue?7176:1167
> eval @ DeployVM.vue?7176:1721
> setTimeout (async)
> updateTemplateParameters @ DeployVM.vue?7176:1716
> updateFieldValue @ DeployVM.vue?7176:1142
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> Vue.$emit @ vue.runtime.esm.js?2b0e:3888
> updateTemplateIso @ TemplateIsoSelection.vue?9200:125
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> Vue.$emit @ vue.runtime.esm.js?2b0e:3888
> onSelectTemplateIso @ TemplateIsoRadioGroup.vue?903a:118
> selected @ TemplateIsoRadioGroup.vue?903a:107
> run @ vue.runtime.esm.js?2b0e:4568
> flushSchedulerQueue @ vue.runtime.esm.js?2b0e:4310
> eval @ vue.runtime.esm.js?2b0e:1980
> flushCallbacks @ vue.runtime.esm.js?2b0e:1906
> Promise.then (async)
> timerFunc @ vue.runtime.esm.js?2b0e:1933
> nextTick @ vue.runtime.esm.js?2b0e:1990
> queueWatcher @ vue.runtime.esm.js?2b0e:4402
> update @ vue.runtime.esm.js?2b0e:4544
> Vue.$forceUpdate @ vue.runtime.esm.js?2b0e:3969
> handleChange @ Checkbox.js?d994:76
> invokeWithErrorHandling @ vue.runtime.esm.js?2b0e:1854
> invoker @ vue.runtime.esm.js?2b0e:2179
> original._wrapper @ vue.runtime.esm.js?2b0e:6917
> vue.runtime.esm.js?2b0e:1888 TypeError: Cannot read property 'length' of undefined
>     at eval (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1446:76)
>     at Proxy.renderList (vue.runtime.esm.js?2b0e:2630)
>     at eval (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1281:49)
>     at Proxy.renderList (vue.runtime.esm.js?2b0e:2651)
>     at Proxy.render (eval at ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"f83824e4-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/views/compute/DeployVM.vue?vue&type=template&id=30bba9f0&scoped=true& (25.js:131), <anonymous>:1258:41)
>     at VueComponent.Vue._render (vue.runtime.esm.js?2b0e:3548)
>     at VueComponent.updateComponent (vue.runtime.esm.js?2b0e:4066)
>     at Watcher.get (vue.runtime.esm.js?2b0e:4479)
>     at Watcher.run (vue.runtime.esm.js?2b0e:4554)
>     at flushSchedulerQueue (vue.runtime.esm.js?2b0e:4310)

File: src/views/compute/DeployVM.vue (hunk 1 above)
Comment (rhtyd) [reply src/views/compute/DeployVM.vue:1811]: use the alert component

**Comment by rhtyd** [comment]
@shwstppr would it be quicker for you to address issues or advise Nicolas? Thanks.
@nvazquez @shwstppr cc @davidjumani - pl advise me accordingly happy to merge.

**Comment by shwstppr** [comment]
Screenshot from 2020-08-19 17-13-51
Added a-alert component and fixed above error

**Comment by rhtyd** [comment]
@nvazquez can you test, I can merge if it works now
@blueorangutan package

**Comment by andrijapanicsb** [comment]
Not good - there should be 3 categories, instead of 1:

image

**Comment by rhtyd** [comment]
Merging this as Abhishek confirms testing this. cc @andrijapanicsb please log further issues, I'm merging this so you can then take latest Primate and carry on vApp backend testing.

**Comment by andrijapanicsb** [comment]
LGTM - both the vApp categories and the questions in each  -  are sorted properly - as in the OVF
thx!

**Pull request merged**